pip_audit/_dependency_source/resolvelib/resolvelib.py

```diff
@@ -72,6 +77,9 @@ def resolve(self, req: Requirement) -> list[Dependency]:
                     SkippedDependency(name=req.name, skip_reason="requirement marked as editable")
                 ]
 
+        # TODO(alex): Figure out a better way to do this
+        self.provider.req_hashes = req_hashes
```

Comment:
The provider gets set when the resolver is constructed (which needs to happen to construct the requirements file). So we obviously can't have the hash information from the requirements file at the point of construction.

We could just create a new `provider` for each `resolve` call... not sure that's really the right answer though.

Reply:
Yeah, that's not ideal. I'll look at this a bit.

IMO, creating a new provider for each resolution isn't the _worst_ idea -- we're reusing HTTP sessions anyways and caching everything, so the provider objects themselves are _relatively_ lightweight (especially in the context of network-dominated execution).

Reply:
Thought about this some more, and I think creating a new provider on each resolution is fine (and preferable for now) -- there's no meaningful overhead due to the shared resources, and it avoids a `None` typestate.

Reply:
[31d7cff](https://github.com/pypa/pip-audit/pull/462/commits/31d7cffc51e5c29bf40861dd6a6f59ff96c8c900)